src/ui/ui_settings/ui_settings_service_for_request.js

```diff
@@ -0,0 +1,20 @@
+import { uiSettingsServiceFactory } from './ui_settings_service_factory';
+
+const BY_REQUEST_CACHE = new WeakMap();
+
+export function getUiSettingsServiceForRequest(server, request, readInterceptor) {
+  if (BY_REQUEST_CACHE.has(request)) {
+    return BY_REQUEST_CACHE.get(request);
+  }
+
+  const adminCluster = server.plugins.elasticsearch.getCluster('admin');
+  const uiSettingsServices = uiSettingsServiceFactory(server, {
+    readInterceptor,
+    callCluster(...args) {
```

Comment:
__question:__ wondering if `callAdminCluster` would make the intention clearer? But if `admin` cluster here is really an obvious thing for other developers then it's fine as is.